Identity/Identity.API/Controllers/IdentityController.cs

```diff
@@ -0,0 +1,143 @@
+using Identity.API.Helpers;
+using Identity.DataAccess;
+using Identity.DataAccess.Models.Entities;
+using Identity.Public;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Net.Http.Headers;
+using System.Security.Claims;
+
+namespace Recipes.API.Controllers;
+
+[ApiController]
+public class IdentityController(IdentityDatabaseContext dbContext) : ControllerBase
+{
+    [HttpGet]
+    [Route("users")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> GetAllRecipes()
+    {
+        IEnumerable<UserEntity> users;
+        try
+        {
+            users = await dbContext.Users.ToListAsync();
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(users);
+    }
+
+    [HttpGet]
+    [Route("user/{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> GetAllRecipes(int id)
+    {
+        User user;
+
+        try
+        {
+            UserEntity? userEntity = await dbContext.Users.FindAsync(id);
+
+            if (userEntity is null)
+                return NotFound();
+
+            user = new User
+            { 
+                Username = userEntity.Username,
+                Email = userEntity.Email
+            };
+        }
+        catch
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError);
+        }
+
+        return Ok(user);
+    }
+
+    [Route("register")]
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<UserEntity>>> RegisterUser(UserCreateRequest userCreateRequest)
+    {
+        UserEntity userToCreate = new UserEntity()
+        {
+            Username = userCreateRequest.Username,
+            PasswordHash = HashingHelper.HashPassword(userCreateRequest.Password),
+            Email = userCreateRequest.Email,
+            Roles = UserRole.Member | UserRole.Admin,
```

Comment:
Let's add `[Flags]` attribute on UserRole.

Reply:
UserRole has been replaced with list of strings as per @BenasB's request.